File: code_samples/r/web_development_middleware.r
# Web Development: Middleware
# AI/ML Training Sample

Middleware <- setRefClass(
    "Middleware",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- Middleware$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")

File: code_samples/java/WebDevelopmentMiddleware.java
/**
 * Web Development: Middleware
 * AI/ML Training Sample
 */
public class WebDevelopmentMiddleware {
    
    private String data;
    
    public WebDevelopmentMiddleware() {
        this.data = "";
    }
    
    public void process(String input) {
        this.data = input;
    }
    
    public String getData() {
        return this.data;
    }
    
    public boolean validate() {
        return this.data != null && !this.data.isEmpty();
    }
    
    public static void main(String[] args) {
        WebDevelopmentMiddleware instance = new WebDevelopmentMiddleware();
        instance.process("example");
        System.out.println("Data: " + instance.getData());
        System.out.println("Valid: " + instance.validate());
    }
}
